### ISSUE TITLE
lint `identity_conversion` could be extended to work with `try_from`/`try_into`

### ISSUE THREAD

**Issue by tmpolaczyk** [rust-lang/rust-clippy#5344]
The following code doesn't warn about the useless conversion from `u8` to `u8`:

```rust
let a = 0u8;
let b = u8::try_from(a).unwrap();
```

[Playground link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=242f3988ce383d57bc2f4cae1758579a)

The [identity_conversion](https://rust-lang.github.io/rust-clippy/master/index.html#identity_conversion) lint currently only supports the traits `Into`, `From`, and `FromIter`. I think it should be simple and useful to extend it to also work with `TryFrom` and `TryInto`.

An alternative would be a new lint to warn on usages of `try_from` when `from` is enough, but I'm not sure how can that be implemented.
